tests/api_test.py

```diff
@@ -1093,6 +1093,42 @@ def vjp(x_tangent):
     b = np.dot(a + np.eye(a.shape[0]), real_x)
     print(gf(a, b))  # doesn't crash
 
+  def test_vmap_in_axes_tree_prefix_error(self):
+    # https://github.com/google/jax/issues/795
+    jtu.check_raises_regexp(
+        lambda: api.vmap(lambda x: x, in_axes=(0, 0))(np.ones(3)),
+        ValueError, "axes specification must be a tree prefix")
+
+  def test_vmap_objects_issue_183(self):
+    # https://github.com/google/jax/issues/183
+    fun = lambda f, x: f(x)
+    vfun = api.vmap(fun, (None, 0))
+    ans = vfun(lambda x: x + 1, np.arange(3))
+    self.assertAllClose(ans, onp.arange(1, 4), check_dtypes=False)
+
+  def test_vmap_error_message_issue_705(self):
+    # https://github.com/google/jax/issues/705
+    def h(a, b):
+      return np.sum(a) + np.sum(b)
+
+    X = onp.random.randn(10, 4)
+    U = onp.random.randn(10, 2)
+    self.assertRaisesRegex(
+        ValueError,
+        "vmap got inconsistent sizes for array axes to be mapped:\n"
+        "arg 0 has shape \(10, 4\) and axis 0 is to be mapped\n"
+        "arg 1 has shape \(10, 2\) and axis 1 is to be mapped\n"
+        "so\n"
+        "arg 0 has an axis to be mapped of size 10\n"
+        "arg 1 has an axis to be mapped of size 2",
```

Comment:
That won't work because of the second case in the PR comment:

```
ValueError: vmap got inconsistent sizes for array axes to be mapped:
arg 0 has shape (10, 4) and axis 0 is to be mapped
arg 1 has shape (10, 2) and axis 1 is to be mapped
arg 2 has shape (10, 4) and axis 0 is to be mapped
so
args 0, 2 have axes to be mapped of size 10  # <-- HERE
arg 1 has an axis to be mapped of size 2
```

Notice how a line can refer to multiple arguments. That means there would be multiple shapes and in_axes to print out.

Reply:
Oh right, I saw that but immediately forgot about it when I started reviewing :) Maybe expand the test case...